web/src/pages/family/FamilyView.tsx

```diff
@@ -1,229 +1,359 @@
+export const FamilyPage: React.FunctionComponent<Record<string, unknown>> = () => {
+    const { familyId } = useParams()
+    if (!familyId) return <em>No family ID</em>
+
+    return <FamilyView familyId={parseInt(familyId)} />
+}
+
+export const FamilyView: React.FC<IFamilyViewProps> = ({ familyId }) => {
+    const [highlightedIndividual, setHighlightedIndividual] = React.useState<
+        string | null | undefined
+    >()
+
+    const [analysisIdToView, setAnalysisIdToView] = React.useState<number | null | undefined>(null)
+
+    if (!familyId || isNaN(familyId)) return <em>Invalid family ID</em>
 
     const { loading, error, data } = useQuery(GET_FAMILY_INFO, {
-        variables: { family_id: family_ID },
+        variables: { family_id: familyId },
     })
 
-    const onPedigreeClick = React.useCallback(
-        (e: string) => {
-            if (!data) return
-            const indexToSet = Object.entries(data?.family.participants)
-                .map(([, value]) => value.externalId)
-                .findIndex((i) => i === e)
-            if (!activeIndices.includes(indexToSet)) {
-                setActiveIndices([...activeIndices, indexToSet])
-            }
-            setMostRecent(e)
-            const element = document.getElementById(e)
-            if (element) {
-                const y = element.getBoundingClientRect().top + window.pageYOffset - 100
-                window.scrollTo({ top: y, behavior: 'smooth' })
+    if (loading) return <LoadingDucks />
+    if (error) return <>Error! {error.message}</>
+    if (!data) return <>No data!</>
+
+    const sgs = data?.family?.familyParticipants.flatMap((fp) =>
+        fp.participant.samples.flatMap((s) => s.sequencingGroups)
+    )
+    const sgsById = _.keyBy(sgs, (s) => s.id)
+
+    const participantBySgId: { [sgId: string]: any } = data?.family?.familyParticipants.reduce(
+        (acc, fp) => {
+            for (const s of fp.participant.samples) {
+                for (const sg of s.sequencingGroups) {
+                    acc[sg.id] = fp.participant as GraphQlParticipant
+                }
             }
+            return acc
         },
-        [data, activeIndices]
+        {} as { [sgId: string]: GraphQlParticipant }
     )
 
-    const handleTitleClick = (e: React.MouseEvent, itemProps: AccordionTitleProps) => {
-        setMostRecent('')
-        const index = itemProps.index ?? -1
-        if (index === -1) return
-        if (activeIndices.indexOf(+index) > -1) {
-            setActiveIndices(activeIndices.filter((i) => i !== index))
-        } else {
-            setActiveIndices([...activeIndices, +index])
+    const aById: {
+        [id: number]: {
+            id: number
+            timestampCompleted?: any | null
+            type: string
+            sgs: string[]
+            meta?: any | null
+            output?: string | null
+        }
+    } = {}
+
+    for (const fp of data?.family?.familyParticipants) {
```

Comment:
This probably needs to be coalesced to something iterable, at the moment if data or family don't exist this'll error out